runtime/compiler/env/j9method.cpp

```diff
@@ -5480,6 +5480,35 @@ TR_ResolvedJ9Method::isJITInternalNative()
    return isNative() && !isJNINative() && !isInterpreted();
    }
 
+uint32_t
+TR_ResolvedJ9Method::romMethodArgCountAtCallSiteIndex(int32_t callSiteIndex)
+   {
+   J9ROMClass *romClass = romClassPtr();
+   J9SRP                 *namesAndSigs = (J9SRP*)J9ROMCLASS_CALLSITEDATA(romClass);
+   J9ROMNameAndSignature *nameAndSig   = NNSRP_GET(namesAndSigs[callSiteIndex], J9ROMNameAndSignature*);
+   J9UTF8                *signature    = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+
+   U_8 paramBuffer[256];
+   uintptr_t    paramElements;
+   uintptr_t    paramSlots;
+   jitParseSignature(signature, paramBuffer, &paramElements, &paramSlots);
+   return paramElements;
+   }
+
+uint32_t
+TR_ResolvedJ9Method::romMethodArgCountAtCPIndex(int32_t cpIndex)
+   {
+   J9ROMMethodRef *romMethodRef = (J9ROMMethodRef *)(cp()->romConstantPool + cpIndex);
+   J9ROMNameAndSignature *nameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
+   J9UTF8                *signature    = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+
+   U_8 paramBuffer[256];
+   uintptr_t    paramElements;
+   uintptr_t    paramSlots;
+   jitParseSignature(signature, paramBuffer, &paramElements, &paramSlots);
+   return paramElements;
+   }
+
```

Comment:
@AlexeyKhrabrov do you know if this actually needs any JITServer changes? #11331 makes it so that all ROM strings are sent to the server and SRPs are updated. If I understand correctly, this means that we don't need to use `TR_ResolvedJ9JITServerMethod::getROMString` to access these strings anymore.

Reply:
Correct. I'm pretty sure that the `getROMString()` methods (there is another one in `ClientSessionData`) can be removed altogether, and a bunch of methods in `TR_ResolvedJ9JITServerMethod` that use `getROMString()` don't need to be overridden anymore.

Reply:
Thanks, I'll create an issue for that.
@nbhuiyan based on the above comment, we don't need any changes for this, so could you remove the JITServer override of this methods?